src/main/java/entity/room/BlackjackRoom.java

```diff
@@ -0,0 +1,52 @@
+package entity.room;
+
+import java.io.IOException;
+import java.util.List;
+
+import data_access.ApiDataAccess;
+import entity.Deck;
+import entity.player.Player;
+import use_case.ApiDataAccessInterface;
+
+/**
+ * A Blackjack-implementation of the Room interface.
+ */
+public class BlackjackRoom implements Room {
+
+    private final String roomName;
+    private final String gameName;
+    private List<Player> curPlayers;
+    private String curTurnPlayer;
+    private String status;
+    private Deck deck;
+    // TODO: should also contain an attribute to store the bet (and update the constructor).
+
+    public BlackjackRoom(String roomName, String gameName, List<Player> curPlayers) throws IOException {
+        this.roomName = roomName;
+        this.gameName = gameName;
+        this.curPlayers = curPlayers;
+        this.curTurnPlayer = curPlayers.get(0).getPlayerID();
+        this.status = "Initializing";
+        final ApiDataAccessInterface apiDataAccessInterface = new ApiDataAccess();
```

Comment:
Entities should not have a "has-a" relationship with DAIs, it should be the other way around, DAI "has-a" game, then that DAI is created in the main (builder) call of the program and injected into some Interactor(s), which can then instantiate a Game and set it in the DAI.

Reply:
for gameName, I implemented a Generic Factory that tells which room to create depending on the "type" of the game, or the gameName.
As for roomName, you're absolutely right, I should delete this.